src/components/parser/multiStopParser.ts

```diff
@@ -47,3 +51,32 @@ export const multiStopParser = (json: EtaPredictionXml) => {
   console.log(result);
   return result;
 };
+
+export function multiStopUnifier(
+  json: EtaPredictionXml,
+  stopBookmarks: stopBookmarkRedux
+) {
+  const result = multiStopParser(json);
+
+  // phase 2: combine a/c to stop number
+  const unifiedList: stopBookmarkWithEta[] = [];
+  for (const id of stopBookmarks.ids) {
+    unifiedList.push({
+      stopId: stopBookmarks.entities[id].stopId,
+      name: stopBookmarks.entities[id].name,
+      lines: stopBookmarks.entities[id].lines,
+      ttcId: stopBookmarks.entities[id].ttcId,
+      etas: [],
+    });
+  }
```

Comment:
This applies to nearly all `for...of...` loop.

I sincerely suggest we should use `map` instead of declaring an empty array and pushing items into it.

- all looping logic is grouped inside `map`, which is clearer and easier for debugging
- `push` operation is significantly slower than `map` [link](https://www.measurethat.net/Benchmarks/Show/11532/0/push-vs-map#latest_results_block)